test/e2e/framework/util.go

```diff
@@ -5100,3 +5103,24 @@ func DsFromManifest(url string) (*extensions.DaemonSet, error) {
 	}
 	return &controller, nil
 }
+
+// WaitForServerPreferredNamespacedResources waits until server preferred namespaced resources could be successfully discovered.
+func WaitForServerPreferredNamespacedResources(d discovery.DiscoveryInterface, timeout time.Duration) ([]*metav1.APIResourceList, error) {
+	Logf("Waiting up to %v for server preferred namespaced resources to be successfully discovered", timeout)
+	var resources []*metav1.APIResourceList
+	if err := wait.PollImmediate(Poll, timeout, func() (bool, error) {
+		var err error
```

Comment:
Add a log saying we are retrying `ServerPreferredNamespacedResources`. Otherwise the error message at line 5120 will be a red herring.

Reply:
Will do.

Reply:
Done.